locations/method_locations.go

```diff
@@ -48,3 +49,9 @@ func MethodOperationInfo(m *desc.MethodDescriptor) *dpb.SourceCodeInfo_Location
 func MethodSignature(m *desc.MethodDescriptor, index int) *dpb.SourceCodeInfo_Location {
 	return pathLocation(m, 4, int(apb.E_MethodSignature.TypeDescriptor().Number()), index) // MethodDescriptor.options == 4
 }
+
+// MethodOption returns the precise location of the method's option with the given name, if any.
+func MethodOption(m *desc.MethodDescriptor, option string) *dpb.SourceCodeInfo_Location {
+	fd := m.GetMethodOptions().ProtoReflect().Descriptor().Fields().ByName(protoreflect.Name(option))
+	return pathLocation(m, 4, int(fd.Number())) // MethodDescriptor.options == 4
+}
```

Comment:
Instead of taking the option as a string, what do you think of taking just the option integer and having the caller figure out the mapping between name and number?

Reply:
That way, the caller can use the descriptor itself to determine the value, like we do above, e.g.

```go
loc := MethodOption(m, int(apb.E_Http.TypeDescriptor().Number()))
```

Reply:
Done